From 0c76610343c257647029890b7bbafa321678d3eb Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Mon, 23 Nov 2020 14:59:18 +0200
Subject: [PATCH] [SW-25120] goya2: use centralized sp sram init

SP SRAM is initialized in cenralized way on init flow. This is then
initializing it during FIT copy. This way the code is aligned with the
rest of the archs. And there is less time waist since initialization is
done before the wait for a new FIT.

Change-Id: I958d407cdc7e90dd9ef4174d221fadfe2d72c9a3
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/goya2/common.c b/goya2/common.c
index 782422b..84fbe4b 100644
--- a/goya2/common.c
+++ b/goya2/common.c
@@ -353,7 +353,11 @@
 
 }
 
-static void init_sp_sram(void)
+/*
+ * initialize scratch SRAM to avoid ECC exceptions ECC errors are observed due
+ * to byte copy to SP memory which turns to be read-modify-write transaction
+ */
+void preboot_sp_sram_init(void)
 {
 	u32 mem_cfg, cpy_sts;
 	u8 retry = HL_INIT_SP_SRAM_RETRIES;
@@ -440,13 +444,6 @@
 static void fit_copy_arc_image(u64 dst, const void *buf, int len)
 {
 	u32 mem_cfg;
-
-	/*
-	 * initialize scratch SRAM to avoid ECC exceptions
-	 * ECC errors are observed due to byte copy to SP
-	 * memory which turns to be read-modify-write transaction
-	 */
-	init_sp_sram();
 
 	/*
 	 * For ARC images:
